src/Appium.Net/Appium/Service/AppiumLocalService.cs

```diff
@@ -112,7 +127,7 @@ public void Start()
                 throw new AppiumServerHasNotBeenStartedLocallyException(msgTxt, e);
             }
 
-            isLaunched = Ping(InitializationTimeout);
+            isLaunched = PingAsync(InitializationTimeout).Result;
```

Comment:
> It seems to me that if this method throws an exception (e.g. `HttpRequestException` or some timeout exception), then `DestroyProcess` is not called. Not sure if it is an issue.

it will be called since PingAsync catch the exception and set `pinged = false` and we return that bool value.

Reply:
> Use of `.Result` is not very common in .NET libraries. People tend to avoid it unless really needed. If possible, it's good to make the parent method async as well ... which might be an issue for backward compatibility. Just sayin for the sake of completeness.

Start method is `[MethodImpl(MethodImplOptions.Synchronized)]` , it means I can't apply this to an async method.